From 732780abc3fa0895912df17fa84304457e55bb92 Mon Sep 17 00:00:00 2001
From: Noah Huetter <noahhuetter@gmail.com>
Date: Mon, 29 Nov 2021 10:03:48 +0100
Subject: [snruntime] declare putchar weak so it can be overridden

Signed-off-by: Noah Huetter <noahhuetter@gmail.com>
---
 sw/snRuntime/src/start_cluster.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sw/snRuntime/src/start_cluster.c b/sw/snRuntime/src/start_cluster.c
index 885c088..b8c11e7 100644
--- a/sw/snRuntime/src/start_cluster.c
+++ b/sw/snRuntime/src/start_cluster.c
@@ -102,7 +102,7 @@ uint32_t _snrt_barrier_reg_ptr() {
 extern uintptr_t volatile tohost, fromhost;
 
 // Provide an implementation for putchar.
-void snrt_putchar(char character) {
+void __attribute__((weak)) snrt_putchar(char character) {
     volatile struct putc_buffer *buf = &putc_buffer[snrt_hartid()];
     buf->data[buf->hdr.size++] = character;
     if (buf->hdr.size == PUTC_BUFFER_LEN || character == '\n') {
-- 
2.15.1

