From 7394b7ea8f87611db1d16328125e5a3a0b21ae91 Mon Sep 17 00:00:00 2001
From: Noah Huetter <noahhuetter@gmail.com>
Date: Mon, 16 Jan 2023 12:23:31 +0100
Subject: [PATCH 06/16] [snRuntime] Clear clint mutex in startup

---
 sw/snRuntime/src/interrupt.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sw/snRuntime/src/interrupt.c b/sw/snRuntime/src/interrupt.c
index cfb7eaab..868ebd8f 100644
--- a/sw/snRuntime/src/interrupt.c
+++ b/sw/snRuntime/src/interrupt.c
@@ -36,6 +36,8 @@ void irq_m_cluster(uint32_t core_idx);
 //================================================================================
 
 void snrt_int_init(struct snrt_team_root *team) {
+    // TODO: this is highly dangerous in multi-cluster systems and must be removed
+    clint_mutex = 0;
     // Put the clint address in tls for faster access
     clint_p = team->peripherals.clint;
     cl_clint_p = team->peripherals.cl_clint;
-- 
2.16.5

