From 7fa52ebb28d95f6130b09faa749307bdb7ea16b4 Mon Sep 17 00:00:00 2001
From: Noah Huetter <noahhuetter@gmail.com>
Date: Mon, 16 Jan 2023 12:26:22 +0100
Subject: [PATCH 08/16] [snRuntime] Change to hierarchical wakeup

---
 sw/snRuntime/src/start.S | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sw/snRuntime/src/start.S b/sw/snRuntime/src/start.S
index 7a2b13e..d58d21a 100644
--- a/sw/snRuntime/src/start.S
+++ b/sw/snRuntime/src/start.S
@@ -182,6 +182,7 @@ snrt.crt0.init_interrupt:
 
     # Synchronize cores.
 snrt.crt0.pre_barrier:
+    call      _snrt_hier_wakeup
     call      _snrt_cluster_barrier
 
     # Execute the main function.
-- 
2.16.5

