From a197fd7141875e4ebe8daeeee4b8a7e54a864c08 Mon Sep 17 00:00:00 2001
From: Noah Huetter <noahhuetter@gmail.com>
Date: Mon, 16 Jan 2023 12:28:18 +0100
Subject: [PATCH 09/15] [snRuntime] Fix FPU barrier

---
 sw/snRuntime/src/ssr.c | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/sw/snRuntime/src/ssr.c b/sw/snRuntime/src/ssr.c
index f5854bf..ffa275d 100644
--- a/sw/snRuntime/src/ssr.c
+++ b/sw/snRuntime/src/ssr.c
@@ -3,11 +3,15 @@
 // SPDX-License-Identifier: Apache-2.0
 #include "snrt.h"
 
-/// Synchronize the integer and float pipelines.
+/**
+ * @brief Synchronize the integer and float pipelines by creating a dependency in the scoreboard on
+ * `tmp`
+ *
+ */
 void snrt_fpu_fence() {
-    unsigned tmp;
-    asm volatile(
-        "fmv.x.w %0, fa0\n"
-        "mv      %0, %0"
-        : "+r"(tmp)::);
+  unsigned tmp;
+  asm volatile("fmv.x.w %0, fa0\n"
+               "mv      %0, %0"
+               : "+r"(tmp)::);
+
 }
-- 
2.16.5

