From 3480f5b0c2111ae3d8b4bfe7b78c50f453293a45 Mon Sep 17 00:00:00 2001
From: Noah Huetter <noahhuetter@gmail.com>
Date: Mon, 10 Jan 2022 14:10:01 +0100
Subject: [sw/snitch] Fix banshee bootloader to return correct exit code

Signed-off-by: Noah Huetter <noahhuetter@gmail.com>
---
 sw/snRuntime/src/start_banshee.S | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sw/snRuntime/src/start_banshee.S b/sw/snRuntime/src/start_banshee.S
index 4c592c1..a4d0347 100644
--- a/sw/snRuntime/src/start_banshee.S
+++ b/sw/snRuntime/src/start_banshee.S
@@ -54,8 +54,8 @@ _snrt_exit:
     sw        a0, 0(t0)
 
     # load 0 -> a0 to idle-loop after this function
-    li        a0, 0
-1:  ret
+1:  li        a0, 0
+    ret
 
 .set tcdm_start_address_reg, 0x40000000
 .set tcdm_end_address_reg, 0x40000008
-- 
2.15.1

