From 7c51f6f07605e1ac1e5da45c9bb10c9f81fa29d8 Mon Sep 17 00:00:00 2001
From: Noah Huetter <noahhuetter@gmail.com>
Date: Wed, 12 Jan 2022 09:06:27 +0100
Subject: [sw/snitch] AXPY kernel cleanup

Signed-off-by: Noah Huetter <noahhuetter@gmail.com>
---
 sw/snBLAS/src/axpy.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/sw/snBLAS/src/axpy.c b/sw/snBLAS/src/axpy.c
index ccfdcb6..47d5000 100644
--- a/sw/snBLAS/src/axpy.c
+++ b/sw/snBLAS/src/axpy.c
@@ -132,9 +132,6 @@ void axpy_block(uint32_t n, uint32_t tile_n, double *alpha, double *dx, double *
 
 void __attribute__((noinline))
 axpy(uint32_t n, double *alpha, double *dx, double *dy, uint32_t setup_ssr) {
-  uint32_t compute_id = snrt_cluster_compute_core_idx();
-  uint32_t compute_num = snrt_cluster_compute_core_num();
-
   double alpha_load = *alpha;
 
   if (setup_ssr) {
-- 
2.15.1

