--- b/sw/cmake/SnitchUtilities.cmake
+++ a/sw/cmake/SnitchUtilities.cmake
@@ -38,6 +38,12 @@
     add_executable(${ARGV})
     target_link_libraries(${name} ${SNITCH_RUNTIME})
     target_link_options(${name} PRIVATE "SHELL:-T ${LINKER_SCRIPT}")
+    add_custom_command(
+        TARGET ${name}
+        POST_BUILD
+        COMMAND ${CMAKE_OBJCOPY}
+        ARGS -O binary --strip-unneeded $<TARGET_FILE:${name}> $<TARGET_FILE:${name}>.bin
+    )
     add_custom_command(
         TARGET ${name}
         POST_BUILD
