--- b/sw/snBLAS/src/axpy.c
+++ a/sw/snBLAS/src/axpy.c
@@ -140,14 +140,14 @@
                        &l1_dy[element_start + bsel * buf_stride], ni == 0);
         t_stop = read_csr(mcycle);

+        if (ccfg->cycle_start == 0)
+          ccfg->cycle_start = t_start;
+        if (ccfg->max_stmps)
+          ccfg->stmps[--ccfg->max_stmps] = t_start;
+
+        ccfg->cycle_end = t_stop;
+        if (ccfg->max_stmps)
+          ccfg->stmps[--ccfg->max_stmps] = t_stop;
-        // if (ccfg->cycle_start == 0)
-        //   ccfg->cycle_start = t_start;
-        // if (ccfg->max_stmps)
-        //   ccfg->stmps[--ccfg->max_stmps] = t_start;
-
-        // ccfg->cycle_end = t_stop;
-        // if (ccfg->max_stmps)
-        //   ccfg->stmps[--ccfg->max_stmps] = t_stop;

         // switch buffers
         bsel = !bsel;
