diff --git a/sw/snRuntime/CMakeLists.txt b/sw/snRuntime/CMakeLists.txt
index b0c10325..a120348f 100644
--- a/sw/snRuntime/CMakeLists.txt
+++ b/sw/snRuntime/CMakeLists.txt
@@ -89,6 +89,7 @@ set(banshee_snitch_sources
 set(rtl_snitch_sources
     src/platforms/rtl/start_snitch.S
     src/platforms/rtl/putchar.c
+    src/platforms/hero/snitch_hero_support.c
 )
 set(hero_snitch_sources
     src/platforms/hero/start_snitch.S
